app/process/watcher/consensus-message/consensus-message.go

```diff
@@ -1,42 +1,105 @@
 package consensusmessage
 
 import (
+	"errors"
 	"github.com/hashgraph/hedera-sdk-go"
 	hederaClient "github.com/limechain/hedera-eth-bridge-validator/app/clients/hedera"
+	"github.com/limechain/hedera-eth-bridge-validator/app/domain/repositories"
 	"github.com/limechain/hedera-eth-bridge-validator/app/process/watcher/publisher"
 	"github.com/limechain/hedera-watcher-sdk/queue"
 	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+	"strconv"
 	"time"
 )
 
 type ConsensusTopicWatcher struct {
-	client      *hederaClient.HederaClient
-	topicID     hedera.ConsensusTopicID
-	typeMessage string
-	maxRetries  int
+	client           *hederaClient.HederaClient
+	topicID          hedera.ConsensusTopicID
+	typeMessage      string
+	maxRetries       int
+	statusRepository repositories.StatusRepository
+	startTimestamp   string
+	started          bool
 }
 
-func NewConsensusTopicWatcher(client *hederaClient.HederaClient, topicID hedera.ConsensusTopicID, maxRetries int) *ConsensusTopicWatcher {
+func NewConsensusTopicWatcher(client *hederaClient.HederaClient, topicID hedera.ConsensusTopicID, repository repositories.StatusRepository, maxRetries int, startTimestamp string) *ConsensusTopicWatcher {
 	return &ConsensusTopicWatcher{
-		client:      client,
-		topicID:     topicID,
-		typeMessage: "HCS_TOPIC_MSG",
-		maxRetries:  maxRetries,
+		client:           client,
+		topicID:          topicID,
+		typeMessage:      "HCS_TOPIC_MSG",
+		statusRepository: repository,
+		maxRetries:       maxRetries,
+		startTimestamp:   startTimestamp,
+		started:          false,
 	}
 }
 
 func (ctw ConsensusTopicWatcher) Watch(q *queue.Queue) {
 	go ctw.subscribeToTopic(q)
 }
 
+func (ctw ConsensusTopicWatcher) retrieveTimestamp() string {
+	if ctw.startTimestamp == "" {
+		now := time.Now()
+		milestoneTimestamp := strconv.FormatInt(now.Unix(), 10)
+		log.Infof("Proceeding to monitor from current moment [%s]\n", now.String())
+		return milestoneTimestamp
+	}
+	return ctw.startTimestamp
+}
+
 func (ctw ConsensusTopicWatcher) subscribeToTopic(q *queue.Queue) {
-	_, e := hedera.NewMirrorConsensusTopicQuery().
+	var err error
+	milestoneTimestamp := ctw.startTimestamp
+
+	if !ctw.started {
```

Comment:
In programing, there is the concept of "boobs". Boobs is when you have many nested ifs that can be untested by reversing the logic. Unlike real boobs, programming boobs are hard to read and look at and should be restricted and eliminated by reversing the logic.

An example of fixing this boobs issue is by making a method `initConsensusWatcher`.

Instead of nesting ifs, you can reverse the logic and return. Example:
```
func initConsensusWatcher(ctw ConsensusTopicWatcher) {
if milestoneTimestamp != "" {
return
}
log.Warnf("[%s] Starting Timestamp was empty, proceeding to get [timestamp] from database.", ctw.topicID.String())
milestoneTimestamp, err = ctw.statusRepository.GetLastFetchedTimestamp(ctw.topicID.String())
if err == nil || !errors.Is(err, gorm.ErrRecordNotFound) {
return
}
log.Warnf("[%s] Database Timestamp was empty, proceeding with [timestamp] from current moment.", ctw.topicID.String())
milestoneTimestamp = strconv.FormatInt(time.Now().Unix(), 10)
e := ctw.statusRepository.CreateTimestamp(ctw.topicID.String(), milestoneTimestamp)
if e != nil {
log.Fatal(e)
}
}
```

Sorry, a bit hard to write code in here